### PR TITLE
Upgrade to Gradle 7.6

### PR DESCRIPTION
**Pull request by chris-ch**
Upgrade to Gradle 7.6

This upgrade enables using the Gradle Linux Packaging Plugin (ospackage) in order to easily generate RPM and DEB packages for the Freenet REference Daemon on Linux OS.

### CODE REVIEW

**Comment by chris-ch** [comment]
I am currently on working on using a Gradle plugin for linux packaging and upgraded Gradle in the process. I just noticed that there is already an exiting PR about upgrading Gradle: https://github.com/freenet/fred/pull/766

Leaving this one open nevertheless.

**Comment by chris-ch** [comment]
Actually it does not make sense to have a duplicate for that task, so withdrawing the PR.

**Comment by ArneBab** [comment]
@chris-ch reopened, because #766 has a critical issue that makes it unmergeable and no reply by the contributor — your PR does not have that problem (and it is much easier to review).

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+         <trusted-key id="6658dd7f76fcdf7812114f85dc032628e927d006" group="com.android.tools.external.lombok" name="lombok-ast" version="0.2.2"/>
+         <trusted-key id="694621a7227d8d5289699830abe9f3126bb741c1" group="com.google.guava"/>
+         <trusted-key id="90ee19787a7bcf6fd37a1e9180c08b1c29100955" group="com.squareup" name="javawriter" version="2.2.1"/>
+         <trusted-key id="9d23533896a9784703585b6286e02c5a42196ca8" group="org.apache.commons" name="commons-compress" version="1.0"/>
+         <trusted-key id="a5bd02b93e7a40482eb1d66a5f69ad087600b22c" group="org.ow2.asm"/>
+         <trusted-key id="cacfbd4755a2fc78709bdd92be096e29edb8d141" group="net.sf.proguard"/>
+         <trusted-key id="d4c89ea4aaf455fd88b22087efe8086f9e93774e" group="junit" name="junit" version="4.12"/>
+         <trusted-key id="d9e6e4cc8f66b2034995c7192189ca6247f3dee5" group="^com[.]android[.]tools($|([.].*))" regex="true"/>
+         <trusted-key id="e85aed155021af8a6c6b7a4a7c7d8456294423ba" group="org.objenesis" name="objenesis" version="1.0"/>
+         <trusted-key id="fa7929f83ad44c4590f6cc6815c71c0a4e0b8edd" group="net.java.dev.jna"/>
```
Comment (ArneBab) [inline gradle/verification-metadata.xml:31]: Can you comment out these trusted keys for the time being and use sha256 hashes for all artifacts?

The sha256 hashes match the hashes in dependencies.properties, so this makes it possible to ensure that what is distributed by auto-update and what’s used during build is the same.

**Comment by ArneBab** [comment]
@chris-ch can you resolve the conflict (nothing serious) and address the one review question? The rest looks good to me.

**Comment by ArneBab** [comment]
I’m sorry that it took so long until you got feedback here — I had hoped for #766 to get done and since your PR was closed already closed when I saw it, I didn’t look in detail into it.

File: gradle/verification-metadata.xml (hunk 1 above)
Comment (chris-ch) [reply gradle/verification-metadata.xml:31]: Thank you Arne... Actually I didn't manage to get rid of the trusted keys. If I comment them out I get the following error:
Artifact was signed with key 0785b3eff60b1b1bea94e0bb7c25280eae63ebe5 (Oleg Kalnichevski <oleg@ural.ru>, Oleg Kalnichevski <olegk@apache.org>) but this key is not in your trusted key list

Comment (ArneBab) [reply gradle/verification-metadata.xml:31]: Thank you for the try! I just checked whether changing the hash of the jna jar triggers a build error and it does, so this suffices.

We need to validate strictly against sha256 for all the shipped dependencies — all which are in dependencies.properties — to ensure that there’s no deviation between our development environments and what nodes actually run.

**Comment by ArneBab** [comment]
Conflicts in build.gradle resolved and PR merged — thank you very much!

**Pull request merged**